### PR TITLE
add wikipages

### PR DESCRIPTION
**Pull request by curioussavage**
:eyeglasses: @ajacksified

### CODE REVIEW

File: src/endpoints/v1.es6.js [hunk 1]
```diff
@@ -1263,6 +1267,57 @@ class APIv1Endpoint {
...
+      get: function(options={}) {
+        var { uri, options } = this.wiki.buildOptions(options);
+
+        return this.baseGet(uri, options, (body) => {
+          const type = body.type || body.kind;
+          switch (type) {
+            case WikiPage._type:
+              return new WikiPage(body.data).toJSON();
+              break;
+            case 'Listing':
```
Comment (ajacksified) [inline src/endpoints/v1.es6.js]: `Listing._type` perhaps?

File: src/models/wikiPage.es6.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    delete props.content_html;
+
+    if (props.revision_by) {
+      props.revision_by = props.revision_by.data;  
+    }
+    
+    super(props);
+  }
+
+  toJSON () {
```
Comment (ajacksified) [inline src/models/wikiPage.es6.js]: probably makes sense to just put this in the base model

File: src/models/wikiPageListing.es6.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import Base from './base';
+
+class WikiPageListing extends Base {
+  constructor(props) {
+    props.pages = props.data.slice();
```
Comment (ajacksified) [inline src/models/wikiPageListing.es6.js]: you should probably also `delete props.data` so you don't end up with two copies

File: src/models/wikiPageSettings.es6.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import Base from './base';
+
+class WikiPageSettings extends Base {
+  constructor(props) {
+    
+    props.pageEditorsList = props.editors.map((item) => {
+      return item.data;
+    });
```
Comment (ajacksified) [inline src/models/wikiPageSettings.es6.js]: newline after this, and nuke the empty newline after `constructor(props) {`

File: src/models/wikiPageSettings.es6.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  }
+
+  toJSON () {
+    let props = this.props
+    props._type = WikiPageSettings._type;
+
+    return props;
+  }
+
+  static _type = 'wikipagesettings';
```
Comment (ajacksified) [inline src/models/wikiPageSettings.es6.js]: camelCase

**Comment by ajacksified** [comment]
:haircut:

**Comment by ajacksified** [comment]
whoops, :nail_care:

File: src/models/wikiPageListing.es6.js (hunk 3 above)
Comment (curioussavage) [inline src/models/wikiPageListing.es6.js]: its deleted on the next line

**Comment by curioussavage** [comment]
:haircut:

File: src/endpoints/v1.es6.js [hunk 6]
```diff
@@ -686,8 +690,7 @@ class APIv1Endpoint {
           api_type: 'json',
         };
 
-        options.changeSet.forEach((prop) => {
-          options.form[prop] = options.model.get(prop);
+        options.changeSet.forEach((prop) => {options.form[prop] = options.model.get(prop);
```
Comment (ajacksified) [inline src/endpoints/v1.es6.js]: add a newline (formatting), and refactor `=>` into `function` unless you need `this` binding

**Comment by ajacksified** [comment]
little nit, then good to go! :nail_care:

**Comment by ajacksified** [comment]
:fish:

**Pull request merged**